build.cake

```diff
@@ -0,0 +1,243 @@
+// Usings
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+// Arguments
+var target = Argument<string>("target", "Default");
+var source = Argument<string>("source", null);
+var apiKey = Argument<string>("apikey", null);
+var version = Argument<string>("targetversion", null);
+var incremental = Argument<bool>("incremental", false);
+
+// Variables
+var configuration = IsRunningOnWindows() ? "Release" : "MonoRelease";
+var solution = File("./Nancy.sln");
+var sharedAssemblyInfo = File("./SharedAssemblyInfo.cs");
+var nancyVersion = "0.0.0.0";
+
+// Directories
+var output = Directory("build");
+var outputBinaries = output + Directory("binaries");
+var outputPackages = output + Directory("packages");
+var outputNuGet = output + Directory("nuget");
+
+///////////////////////////////////////////////////////////////
+
+Setup(context => 
+{
+  // Parse the Nancy version.
+  var assemblyInfo = ParseAssemblyInfo(sharedAssemblyInfo);
+  nancyVersion = assemblyInfo.AssemblyVersion;
+  Information("Version: {0}", nancyVersion);
+});
+
+///////////////////////////////////////////////////////////////
+
+Task("Clean")
+  .Does(() =>
+{
+  // Clean artifact directories.
+  CleanDirectories(new DirectoryPath[] {
+    output, outputBinaries, outputPackages, outputNuGet
+  });
+
+  if(!incremental) {
+    // Clean output directories.
+    CleanDirectories("./src/**/" + configuration);
+    CleanDirectories("./test/**/" + configuration);
+    CleanDirectories("./samples/**/" + configuration);
+  }
+});
+
+Task("Restore-NuGet-Packages")
+  .Description("Restores NuGet packages")
+  .Does(() =>
+{
+  var settings = new DotNetCoreRestoreSettings
+  {
+    Verbose = false,
+    Verbosity = DotNetCoreRestoreVerbosity.Warning
+  };
+  
+  DotNetCoreRestore("./src", settings);
+  DotNetCoreRestore("./test", settings);
+});
+
+Task("Compile")
+  .Description("Builds the solution")
+  .IsDependentOn("Clean")
+  .IsDependentOn("Restore-NuGet-Packages")
+  .Does(() =>
+{
+  var projects = GetFiles("./**/*.xproj");
+  foreach(var project in projects)
+  {
+    DotNetCoreBuild(project.GetDirectory().FullPath, new DotNetCoreBuildSettings {
+      Configuration = configuration,
+      Verbose = false
+    });
+  }
+});
+
+Task("Test")
+  .Description("Executes xUnit tests")
+  .WithCriteria(false)
+  .IsDependentOn("Compile")
+  .Does(() =>
+{
+  var projects = GetFiles("./test/**/*.xproj")
+    - GetFiles("./test/**/Nancy.ViewEngines.Razor.Tests.Models.xproj");
+    
+  foreach(var project in projects)
+  {
+    DotNetCoreTest(project.GetDirectory().FullPath, new DotNetCoreTestSettings {
+      Configuration = configuration
+    });
+  }
+});
+
+Task("Publish")
+  .Description("Gathers output files and copies them to the output folder")
+  .IsDependentOn("Compile")
+  .Does(() =>
+{
+  var files = GetFiles("src/**/bin/" + configuration + "/net452/*.dll")

```

Comment:
Ideally we'd want the `netstandard` (and any other target we have) to also be in here.